src/backend/core/services/lobby_service.py

```diff
@@ -0,0 +1,271 @@
+            status = LobbyParticipantStatus.WAITING
+
+        livekit_config = None
+        if status == LobbyParticipantStatus.ACCEPTED:
+            # wrongly named, contains access token to join a room
+            livekit_config = utils.generate_livekit_config(
+                room_id=room_id, user=user, username=username
+            )
+
+        return status, livekit_config
+
+    def refresh_waiting_status(self, room_id: str, participant_id: str):
+        """Refresh timeout for waiting participant.
+
+        Extends the waiting period for a participant to maintain their position
+        in the lobby queue. Automatic removal if the participant is not
+        actively checking their status.
+        """
+        cache.touch(
+            self._get_cache_key(room_id, participant_id), settings.LOBBY_WAITING_TIMEOUT
+        )
+
+    def enter(self, room_id: str, participant_id: str, username: str) -> None:
+        """Add participant to waiting lobby.
+
+        Create a new participant entry in waiting status and notify room
+        participants of the new entry request.
+
+        Raises:
+            LobbyNotificationError: If room notification fails
+        """
+        participant = LobbyParticipant(
+            status=LobbyParticipantStatus.WAITING, username=username, id=participant_id
+        )
+
+        cache_key = self._get_cache_key(room_id, participant_id)
+        cache.set(
+            cache_key,
+            participant.to_dict(),
+            timeout=settings.LOBBY_WAITING_TIMEOUT,
+        )
+
+        try:
+            self.notify_participants(room_name=room_id)
+        except LobbyNotificationError:
+            # If room not created yet, there is no participants to notify
+            pass
+
+    def check_status(self, room_id: str, participant_id: str) -> LobbyParticipantStatus:
+        """Check participant's current status in the lobby."""
+
+        cache_key = self._get_cache_key(room_id, participant_id)
+        data = cache.get(cache_key)
+
+        if not data:
+            return LobbyParticipantStatus.UNKNOWN
+
+        try:
+            participant = LobbyParticipant.from_dict(data)
+            return participant.status
+
+        except LobbyParticipantParsingError:
+            logger.error("Corrupted participant data found and removed: %s", cache_key)
+            cache.delete(cache_key)
+            return LobbyParticipantStatus.UNKNOWN
+
+    def list_waiting_participants(self, room_id: str) -> List[dict]:
+        """List all waiting participants for a room."""
+
+        pattern = self._get_cache_key(room_id, "*")
+        keys = cache.keys(pattern)
+
+        if not keys:
+            return []
+
+        data = cache.get_many(keys)
+
+        return [
+            raw_participant
+            for raw_participant in data.values()
+            if raw_participant
+            and raw_participant.get("status") == LobbyParticipantStatus.WAITING.value
+        ]
```

Comment:
It's the only method you are not dealing with the `LobbyParticipant` object and so not validating the data coming from the cache.